rust/tools/authorization-logic/raksha_examples/VisionPipelineSimple

```diff
@@ -0,0 +1,146 @@
+// limitations under the License.
+//-----------------------------------------------------------------------------
+
+"EndUser" says {
+    // The user owns two tags that reprsent their data:
+    //      - Raw video from the camera
+    //      - Text describing a product that was in frame
+    // The user gives different permissions for how each can be used (more is 
+    // allowed with the text). Because the transformation of data affects what 
+    // can be done with it, it is also controlled by the policy.
+    ownsTag("raw_video_tag").
+    ownsTag("product_id_tag").
+    
+    // Only local, in-memory computation can be done on raw_video_tags, but
+    // product IDs can be sent to remote services. The EndUser trusts the TPPR 
+    // to grant just these permissions to specific relevant pieces of software 
+    // (the variable principal, SoftwareX).
+    "ThirdPartyPrivacyReviewer" canSay SoftwareX
+        may("in_memory_computation","raw_video_tag") :- isPrincipal(SoftwareX).
+    "ThirdPartyPrivacyReviewer" canSay SoftwareX
+        may("egress_to_shopping_service", "product_id_tag") :- isPrincipal(SoftwareX).
+      
+    // These lines say the end user trusts (i.e., delegates to) the TPPR to
+    // faithfully say when a data transformation from raw camera video to the
+    // name of a product has taken place. The TPPR can remove the "raw_video"
+    // tag only when the "product_id_tag" is added.
+    "ThirdPartyPrivacyReviewer" canSay declassifies(pathX, "raw_video_tag") :-
+        hasTag(pathX, "product_id_tag"),  
+        isAccessPath(pathX).
+
+    // The user trusts TPPR to faithfully apply its tags.
+    "ThirdPartyPrivacyReviewer" canSay hasTag(x, y) :- 
+        isAccessPath(x), isTag(y).
+
+    // These are boilerplate that just mention that these principals exist.
+    // Ideally we'll get rid of this in the future.
+    isPrincipal("DataSource").
+    isPrincipal("ImageDetector").
+    isPrincipal("ImageSelector").
+    isPrincipal("ProductIdOutput").
+}
+
+"ThirdPartyPrivacyReviewer" says {
+    // TPPR is aware of the specific software modules in the pipeline,
+    // and grants them the needed permissions.
+    "ImageDetector" may("in_memory_computation", "raw_video_tag").
+    "ImageSelector" may("in_memory_computation", "raw_video_tag").
+    "ProductIdOutput" may("egress_to_shopping_service",
+        "product_id_tag").
+
+    // Remove raw_video_tag and add product_id_tag
+    "ImageSelector" canSay declassifies(pathX, "raw_video_tag") :-
+        hasTag(pathX, "product_id_tag"), 
+        isAccessPath(pathX).
+    "ImageSelector" canSay hasTag(pathX, "product_id_tag") :- 
+        isAccessPath(pathX).
+}
+
+"ImageDetector" says "ImageDetector"
+    will("in_memory_computation", "raw_video_tag").
+"ImageSelector" says {
+    "ImageSelector" will("in_memory_computation", "raw_video_tag").
+    declassifies("is_select_image_id", "raw_video_tag").
+    hasTag("is_select_image_id", "product_id_tag").
+}
```

Comment:
Can we replace this with a particle spec? If so, let us document what that would look like as a comment here.

Reply:
Maybe if we extend the syntax of particle specs we could. I added a comment.

By the way this does correspond to a manifest which is almost the same as the one we had for the un-simplified version.  The only difference is that for the simpler version there is one fewer tag. I checked in a separate .arcs file with that change as `third_party/arcs/examples/VisionPipelineSimple.arcs`